ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/SniHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.MISDIRECTED_REQUEST;
+
+import io.confluent.ksql.rest.Errors;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+public class SniHandler implements Handler<RoutingContext> {
+
+  public SniHandler() {
+  }
+
+  @Override
+  public void handle(final RoutingContext routingContext) {
+    if (routingContext.request().isSSL()) {
```

Comment:
Does this not need null checks?

Reply:
i don't think it's necessary, we don't do it anywhere else in the code. See https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/LoggingHandler.java#L88